### PR TITLE
Update CBMC proofs to use the latest starter kit.

### PR DESCRIPTION
**Pull request by markrtuttle**
This pull requests updates the CBMC proofs to use the latest versions of the proof starter kit and litani. With these changes, the CBMC proofs are built and checked identically at the command prompt and in continuous integration.

The first five commits in this pull request

* Update CBMC litani and starter kit submodules.
* Update CBMC proof files from starter kit.
* Remove CBMC prepare.py needed only for legacy CI.
* Update definitions in CBMC proof Makefiles

Then

* One commit updates Makefile-jobs.common to work with the latest version of the starter kit that now depends on litani: it replaces the sed command in Makefile-jobs.common with a litani command invoking sed.
* One commit fixes a too-big constant in the Jobs_Describe Makefile that was violating a check in a jobs header file.

There remains one loop-unwinding error in Jobs_Describe to discuss.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sarenameas — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Review by sarenameas — DISMISSED** [review]

File: test/cbmc/proofs/Jobs_Describe/Makefile [hunk 1]
```diff
@@ -7,8 +7,8 @@ HARNESS_FILE=Jobs_Describe_harness
 # to provide 100% coverage and to reduce unwindings
 THINGNAME_MAX_LENGTH=6
 EXCEED_THINGNAME_MAX_LENGTH=7
-JOBID_MAX_LENGTH=70
-EXCEED_JOBID_MAX_LENGTH=71
+JOBID_MAX_LENGTH=64
+EXCEED_JOBID_MAX_LENGTH=65
```
Comment (dan4thewin) [inline test/cbmc/proofs/Jobs_Describe/Makefile:11]: Oops, I think I committed that 70 and 71 by accident.  The earlier version had much smaller values, like 4 and 5.  Whichever values we use, the UNWINDSET below needs to change if EXCEED_JOBID_MAX_LENGTH is greater than EXCEED_THINGNAME_MAX_LENGTH.

File: test/cbmc/proofs/Makefile-project-defines [hunk 2]
```diff
@@ -13,6 +13,18 @@
 # Makefile.common.
 ################################################################
 
+# Name of this proof project, displayed in proof reports. For example,
+# "s2n" or "Amazon FreeRTOS". For projects with multiple proof roots,
+# this may be overridden on the command-line to Make, for example
+#
+# 	  make PROJECT_NAME="FreeRTOS MQTT" report
+#
+PROJECT_NAME = "FreeRTOS coreJobs"
```
Comment (dan4thewin) [inline test/cbmc/proofs/Makefile-project-defines:22]: I think "coreJobs", though it has a nice right to it, isn't the right name.

**Review by dan4thewin — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/Makefile-project-defines (hunk 2 above)
Comment (markrtuttle) [reply test/cbmc/proofs/Makefile-project-defines]: Changed to "Jobs library"

File: test/cbmc/proofs/Jobs_Describe/Makefile (hunk 1 above)
Comment (markrtuttle) [reply test/cbmc/proofs/Jobs_Describe/Makefile]: Changed to 6 and 7 like thingname.

**Review by dan4thewin — APPROVED** [review]

**Review by sarenameas — APPROVED** [review]

**Pull request merged**